dynawo/sources/Modeler/Common/DYNSubModel.h

```diff
@@ -730,6 +730,14 @@ class SubModel {
    */
   void addMessage(const std::string& message);
 
+  /**
+   * @brief getter to tell whether or not simulation generates a timeline
+   * @return true when simulation generates a file resuming the timeline; false when timeline has not been asked from jobs file
+   */
+  inline bool hasTimeline() const {
+    return timeline_ ? true : false;
```

Comment:
There is an API for shared_ptr which directly returns this status (use_count == 0)

Reply:
tell me more... You mean I may write "return timeline_.use_count() ? true : false ; }".  The compiler will do the job as I wrote it, no (?)

Reply:
you can write directly "return timeline_.use_count() > 0;" or even "return timeline_" It is for readability. There is never need to use "(sth...) ? true : false" because it means that you can just use "(sth)"

Reply:
You are right, I did not think of it :( let's keep it this way. I have enough trouble pushing changes... I'm not sure "return timeline_;" works since timeline_ is not of bool type, it's a class. Compiler may be smart enough to turn it implicitly to a bool since it is a bool function; anyhow it won't run faster if I change if, for sure.

Reply:
I agree with Florent, so as anyway there will be an update let's update this.

Reply:
I can change "return timeline_ ? true : false;" to "return timeline_.use_count() >0;" . In the first cas the compiler and boost chooses the best way to treat the line. In the second Florent tells the compiler what to do. My opnion is to always let compiler do optimization. The third way "return timeline_;" does something else.